src/components/TodoItem.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+    <li>
+        <div v-if="!isEditing">
+            {{ todo.message }}
+            <button id="edit" @click="edit">Edit</button>
+            <button id="delete" @click="delToDo">Delete</button>
+        </div>
+        <div v-else>
+            <input v-model="temp" />
+            <button id="save" @click="save">Save</button>
+            <button @click="cancel">Cancel</button>
+        </div>
+    </li>
+</template>
+
+<script>
+    export default {
+        name: "TodoItem",
+        props: ["todo", "index"],
+        data() {
+            return {
+                isEditing: this.todo.isEditing,
+                temp: ""
+            };
+        },
+        methods: {
+            edit: function() {
+                this.temp = this.todo.message;
+                this.isEditing = true;
+            },
+            delToDo: function() {
+                this.isEditing = false;
+                this.$emit("deleteEntry", this.index);
+                // this.$parent.deleteEntry(this.index);
+            },
+            save: function() {
+                this.isEditing = false;
+                this.todo.message = this.temp;
+                this.$emit("saveEntry", this.temp, this.index);
```

Comment:
```suggestion
                this.$emit("saveEntry", this.todo , this.index);
```

It shouldn't be the List's job to put the object together.

Reply:
Using `this.todo` will throw an exception for me
`TypeError: Converting circular structure to JSON`
but we will change it to:
```suggestion
                this.$parent.saveEntry(this.todo, this.index);
```

Reply:
We changed it in the next commit.